src/components/Badge/__stories__/Badge.stories.mdx

```diff
@@ -0,0 +1,213 @@
+import Badge from "../Badge";
+import { ArgsTable, Story, Canvas, Meta } from "@storybook/addon-docs";
+import person from "./assets/person.png";
+import { ExternalPage, WhatsNew } from "../../Icon/Icons";
+import Link from "../../Link/Link";
+import Button from "../../Button/Button";
+import Flex from "../../Flex/Flex";
+import Avatar from "../../Avatar/Avatar";
+import { createStoryMetaSettingsDecorator } from "../../../storybook";
+import { createComponentTemplate, StoryDescription } from "vibe-storybook-components";
+import { BUTTON, COUNTER, LINK } from "../../../storybook/components/related-components/component-description-map";
+
+export const metaSettings = createStoryMetaSettingsDecorator({
+  component: Badge,
+  enumPropNamesArray: ["color", "position", "type"],
+  iconPropNamesArray: [],
+  actionPropsArray: []
+});
+
+<Meta title="Feedback/Badge" component={Badge} argTypes={metaSettings.argTypes} decorators={metaSettings.decorators} />
+
+<!--- Component template -->
+
+export const badgeTemplate = createComponentTemplate(Badge);
+
+<!--- Component documentation -->
+
+# Badge
+
+- [Overview](#overview)
+- [Props](#props)
+- [Variants](#variants)
+- [Do’s and don’ts](#dos-and-donts)
+- [Use cases and examples](#use-cases-and-examples)
+- [Related components](#related-components)
+- [Feedback](#feedback)
+
+## Overview
+
+Badge component is responsible for positioning an indicator/counter on a child component
+
+<Canvas>
+  <Story name="Overview" args={{ children: <Button leftIcon={WhatsNew}>What's new</Button> }}>
+    {badgeTemplate.bind({})}
+  </Story>
+</Canvas>
+
+## Props
+
+<ArgsTable story="Overview" />
+
+## Variants
+
+### States
+
+Badge can be of type Indicator or type Counter
+
+<Canvas>
+  <Story name="States">
+    <Flex gap={Flex.gaps.LARGE} style={{ flex: 1 }} justify={Flex.justify.START} align={Flex.align.START}>
+      <StoryDescription description="Indicator" vertical align={Flex.align.START}>
+        <Badge>
+          <Button leftIcon={WhatsNew}>What's new</Button>
+        </Badge>
+      </StoryDescription>
+      <StoryDescription description="Counter" vertical align={Flex.align.START}>
+        <Badge type={Badge.types.COUNTER}>
```

Comment:
Since it's adding a counter number by default, it might be better to have this example with the `counterProps`.
<img width="176" alt="image" src="https://github.com/mondaycom/monday-ui-react-core/assets/6093192/d7f6acf6-f187-444e-bb33-a7a1377a137c">